Clarify the behavior of PATCH

kperk103 opened the pull request:
This adds documentation on the behavior of PATCH compared
to the behavior of PUT.

This is my first time writing documentation for something like this so I'd love to get some feedback on it!

Closes #52

Commit:
Clarifies the behavior of PATCH

This adds documentation on the behavior of PATCH compared
to the behavior of PUT.

laurelmay commented on openapi.yaml:
````diff
@@ -55,10 +61,14 @@ info:
         }
       }
       ```
+      In this example, the title is being updated and the version is being added to the list of revisions.
````
I think we may want additional emphasis on the _added_ part here. We want to emphasize that no other changes should be made to the `metadata` object as a whole or the contents of the `revisions` list.

Commit:
Add more detail to the example description

This adds more detail to the description of the given
example. It places emphasis on the idea that the
list will simply add the version item and the metadata
object will still contain its unedited contents
as well.

laurelmay reviewed (approved):
Thanks @kperk103! I think this updated language should be really helpful context for the goals of our `PATCH` operation. Thank you!

laurelmay commented on openapi.yaml:
```diff
@@ -40,6 +40,12 @@ info:
     </details>
 
     ## Partial Payloads in PATCH Requests
+    A `PATCH` should be used when an existing object needs a partial modification. It accepts a partial
```
```suggestion

    A `PATCH` should be used when an existing object needs a partial modification. It accepts a partial
```

Commit:
Minor formatting update for spec markdown
